packages/datetime/src/datePicker.tsx

```diff
@@ -268,6 +301,8 @@ export class DatePicker extends AbstractComponent<IDatePickerProps, IDatePickerS
             }
         }
 
+        Utils.safeInvoke(this.props.dayPickerProps.onMonthChange, value);
```

Comment:
tricky question that I've never been able to answer: should we invoke overridden prop handlers like this _before_ or _after_ our own logic? 

this PR is internally inconsistent: `onDayClick` is called immediately, but the others are called somewhere in the middle.